src/modules/governance/proposal/VotersListContainer.tsx

```diff
@@ -0,0 +1,187 @@
+  twitterAvatar: string | null;
+  twitterHandle: string | null;
+  twitterName: string | null;
+  vote: 0 | 1; // 0 for nay, 1 for yae
+  votingHistory: VoterVoteHistoryItem[]; //(2) [{…}, {…}]
+  votingPower: number; // the amount of voting power the user has - 2.6209877046554566
+  votingWeight: number; // the % that a single user contributes to the total - 1.6381173154096605e-7
+};
+
+type GovernanceProposalTopVotersResponse = [GovernanceVoter[], GovernanceVoter[]];
+
+export type VotersData = {
+  yaes: GovernanceVoter[];
+  nays: GovernanceVoter[];
+  combined: GovernanceVoter[];
+};
+
+const sortByVotingPower = (a: GovernanceVoter, b: GovernanceVoter) => {
+  return a.votingPower < b.votingPower ? 1 : a.votingPower > b.votingPower ? -1 : 0;
+};
+
+export const VotersListContainer = (props: VotersListProps): JSX.Element => {
+  const { proposal } = props;
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState<boolean>(false);
+  const [voters, setVoters] = useState<VotersData>();
+  const [votersModalOpen, setVotersModalOpen] = useState(false);
+
+  const votersUrl = `${process.env.NEXT_PUBLIC_API_BASEURL}/data/proposal-top-voters`;
+  const queryParams = `?proposal=${proposal.id}`;
+
+  useEffect(() => {
+    const getVoterInfo = async () => {
+      if (error) setError(false);
+
+      try {
+        // Get proposal voters data
+        const resp = await fetch(votersUrl + queryParams);
+
+        if (resp.ok) {
+          const [yaes, nays]: GovernanceProposalTopVotersResponse = await resp.json();
+          // Transform data for UI, sort by highest voting power
+          const yesVoters: GovernanceVoter[] = yaes.map((v: GovernanceVoter) => {
+            const proposalVote = v.votingHistory.find(
+              (h) => h.proposal.id === proposal.id.toString()
+            );
+            return {
+              ...v,
+              vote: 1,
+              proposalVotingPower: proposalVote?.votingPower ?? 0,
+            };
+          });
+          const noVoters: GovernanceVoter[] = nays.map((v: GovernanceVoter) => {
+            const proposalVote = v.votingHistory.find(
+              (h) => h.proposal.id === proposal.id.toString()
+            );
+            return {
+              ...v,
+              vote: 0,
+              proposalVotingPower: proposalVote?.votingPower ?? 0,
+            };
+          });
+          const votersData: VotersData = {
+            yaes: yesVoters.sort(sortByVotingPower),
+            nays: noVoters.sort(sortByVotingPower),
+            combined: yesVoters.concat(noVoters).sort(sortByVotingPower),
+          };
+          setVoters(votersData);
+        } else {
+          setError(true);
+        }
+      } catch (e: unknown) {
+        console.error(e);
+        setError(true);
+      }
+      setLoading(false);
+    };
+    getVoterInfo();
+  }, [proposal]); /* eslint-disable-line react-hooks/exhaustive-deps */
```

Comment:
You could add a const for the proposal Id and just add that as a dependency for this `useEffect` to prevent firing off a network request on every re-render.

Reply:
Correct me if I'm wrong, but if having the page open on an active proposal, and a new vote comes in, then we won't fetch the new voters in a live view.  The user would have to reload the page to see the new votes.  In the next.js page, there is a polling to fetch the live proposal information every 30s.  This is why I went with the overall proposal as the trigger for the `useEffect`, so that voter information could be populated in realtime every 30s.

Reply:
Yeah, that makes sense